src/screens/Vault/SigningDeviceDetails.tsx

```diff
@@ -412,21 +411,28 @@ function SigningDeviceDetails({ route }) {
           navigation.goBack();
           throw new Error('Please import the vault before signing');
         }
+
+        if (
+          parseInt(changeAddressIndex) >
+          activeVault.specs.nextFreeChangeAddressIndex + CHANGE_INDEX_THRESHOLD
+        )
+          throw new Error('Change index is too high.');
+
         const psbtWithGlobalXpub = await getPsbtForHwi(serializedPSBT, activeVault);
         serializedPSBT = psbtWithGlobalXpub.serializedPSBT;
+        receiverAddresses = findChangeFromReceiverAddresses(
```

Comment:
This will only work if the device used is in the SignersReqVault. But we should do change detection for all device types...